### PR TITLE
vendor: github.com/docker/docker v27.0-dev (018d93decfb5)

### PR DESCRIPTION
**Pull request by vvoland**
full diff: https://github.com/docker/docker/compare/1a1f3cff45ec...018d93decfb5

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

### CODE REVIEW

**Comment by thaJeztah** [comment]
Hm... looks like the linter isn't happy because we're copying locks somewhere 😬

**Comment by vvoland** [comment]
Hmm right, `Client` became non-copyable due to the `atomic.Bool` (and implictily `sync.Mutex`) being added to the struct, and our `fakeClient` actually embeds the `Client` by value.

File: cli/command/container/client_test.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 type fakeClient struct {
-	client.Client
+	client.APIClient
```
Comment (thaJeztah) [inline cli/command/container/client_test.go:18]: I guess one of the tests is now trying to call the interface somewhere 🤔 (which panics);


```
#20 61.64 === FAIL: cli/command/container TestNewCreateCommandWithContentTrustErrors (0.00s)
#20 61.64 panic: runtime error: invalid memory address or nil pointer dereference [recovered]
#20 61.64 	panic: runtime error: invalid memory address or nil pointer dereference
#20 61.64 [signal SIGSEGV: segmentation violation code=0x1 addr=0x178 pc=0xc91b5e]
#20 61.64 
#20 61.64 goroutine 77 [running]:
#20 61.64 testing.tRunner.func1.2({0xd84e20, 0x16e8a50})
#20 61.64 	/usr/local/go/src/testing/testing.go:1545 +0x238
#20 61.64 testing.tRunner.func1()
#20 61.64 	/usr/local/go/src/testing/testing.go:1548 +0x397
#20 61.64 panic({0xd84e20?, 0x16e8a50?})
#20 61.64 	/usr/local/go/src/runtime/panic.go:914 +0x21f
#20 61.64 github.com/docker/cli/cli/command/container.(*fakeClient).DaemonHost(0xec5ac1?)
#20 61.64 	<autogenerated>:1 +0x1e
#20 61.64 github.com/docker/cli/cli/command/container.runCreate({0x103af40, 0x1740aa0}, {0x1043c28?, 0xc0003a0240?}, 0x0?, 0xc0003be000, 0xc0003f4b00)
#20 61.64 	/go/src/github.com/docker/cli/cli/command/container/create.go:88 +0xee
#20 61.64 github.com/docker/cli/cli/command/container.NewCreateCommand.func1(0xc0003ae100?, {0xc0003f2320?, 0x4?, 0xec2c34?})
#20 61.64 	/go/src/github.com/docker/cli/cli/command/container/create.go:58 +0x125
#20 61.64 github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc000004300, {0xc0003f2190, 0x1, 0x1})
#20 61.64 	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:985 +0xabc
#20 61.64 github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000004300)
#20 61.64 	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:1117 +0x3ff
#20 61.64 github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
#20 61.64 	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:1041
#20 61.64 github.com/docker/cli/cli/command/container.TestNewCreateCommandWithContentTrustErrors(0x0?)
#20 61.64 	/go/src/github.com/docker/cli/cli/command/container/create_test.go:240 +0x1ec
#20 61.64 testing.tRunner(0xc0003dc340, 0xf39900)
#20 61.64 	/usr/local/go/src/testing/testing.go:1595 +0xff
#20 61.64 created by testing.(*T).Run in goroutine 1
#20 61.64 	/usr/local/go/src/testing/testing.go:1648 +0x3ad
```

Comment (vvoland) [reply cli/command/container/client_test.go]: https://github.com/docker/cli/pull/5181

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**